public/services/todo-api.js

```diff
@@ -0,0 +1,82 @@
+const URL = '/api';
+
+async function fetchWithError(url, options) {
```

Comment:
your fetch with error is missing a few things. 
use this instead
`async function fetchWithError(url, options) {
    if (token) {
        options = options || {};
        options.headers = options.headers || {};
        options.headers.Authorization = token;
    }
    const response = await fetch(url, options);
    const data = await response.json();

    if (response.ok) {
        return data;
    }
    else {
        throw data.error;
    }
}`

Reply:
ahh thank you sir!  I;m not sure I would've every fond that without your help.  I appreciate it!